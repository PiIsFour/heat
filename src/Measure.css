.Measure {
	border: 2px solid gray;
	padding: 1em;
	margin: 1em;
	text-align: center;
}

.Measure > input {
	text-align: right;
	margin-left: .5em;
	margin-right: .5em;
	max-width: 5em;
}
