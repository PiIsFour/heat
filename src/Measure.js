import React, {Component} from 'react'
import './Measure.css'

class Measure extends Component {
	constructor (props) {
		super(props)
		this.handleValueUpdate = this.handleValueUpdate.bind(this)
		this.handleInputUpdate = this.handleInputUpdate.bind(this)
	}
	handleValueUpdate (event) {
		const {valueChanged, name} = this.props.state
		valueChanged(name, event.target.value)
	}
	handleInputUpdate (event) {
		const {inputToggled, name, input} = this.props.state
		inputToggled(name, !input)
	}
	render () {
		const {unit, input, calculated, symbol, value, longname, formul} = this.props.state
		const style = {}
		if (!input && !calculated) {
			style.backgroundColor = 'gray'
		}
		return <div className="Measure" style={style}>
			<div>{longname}</div>
			<div>
				<input className="input" type="checkbox" checked={input} disabled={calculated} onChange={this.handleInputUpdate}/>
				<math xmlns="http://www.w3.org/1998/Math/MathML">{symbol}</math>
				<input className="value" type="text" readOnly={!input} disabled={!input && !calculated} value={calculated ? value.toFixed(2) : value} onChange={this.handleValueUpdate}/>
				<span className="unit">{unit}</span>
			</div>
			<div>{formul}</div>
		</div>
	}
}

export default Measure
