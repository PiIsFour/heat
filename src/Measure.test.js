import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Measure from './Measure'

describe('Measure', () => {
	const state = {
		name: 't w in',
		unit: 'kg',
		input: false,
		value: 42,
		symbol:
			<math xmlns="http://www.w3.org/1998/Math/MathML">
				<mrow>
					<msub><mi>T</mi><mi>W in</mi></msub>
				</mrow>
			</math>,
		valueChanged: jest.fn(),
		inputToggled: jest.fn()
	}
	const component = shallow(
		<Measure state={state}/>
	)

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<Measure state={state}/>
		)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('contains the form elements', () => {
		expect(component.find('input.input')).toHaveLength(1)
		expect(component.find('input.value')).toHaveLength(1)
	})

	it('contains the unit', () => {
		expect(component.text()).toContain(state.unit)
	})

	it('updates the form when keys are pressed', () => {
		const updateKey = '12'
		component.instance().handleValueUpdate({ target: { value: updateKey } })
		expect(state.valueChanged).toBeCalled()
	})

	it('updates the form when input is toggled', () => {
		component.instance().handleInputUpdate()
		expect(state.inputToggled).toBeCalled()
	})
})
