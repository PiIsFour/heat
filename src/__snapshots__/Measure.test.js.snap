// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Measure renders and matches our snapshot 1`] = `
<div
  className="Measure"
  style={
    Object {
      "backgroundColor": "gray",
    }
  }
>
  <input
    checked={false}
    className="input"
    disabled={undefined}
    onChange={[Function]}
    type="checkbox"
  />
  <math
    xmlns="http://www.w3.org/1998/Math/MathML"
  >
    <mrow>
      <msub>
        <mi>
          T
        </mi>
        <mi>
          W in
        </mi>
      </msub>
    </mrow>
  </math>
  <input
    className="value"
    disabled={true}
    onChange={[Function]}
    readOnly={true}
    type="text"
    value={42}
  />
  <span
    className="unit"
  >
    kg
  </span>
  <div />
</div>
`;
